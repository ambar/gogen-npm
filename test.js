const {mock} = require('gogen')

const directory = 'dist'
const answers = {
  description: 'superb',
  devDeps: [],
}

describe('template', () => {
  it('creates with defaults', async () => {
    const {files, readFile} = await mock('.', directory, {
      answers,
    })
    expect(files).toMatchSnapshot()
    expect(JSON.parse(readFile('package.json'))).toMatchSnapshot()
  })

  it('creates with jest', async () => {
    const {readFile} = await mock('.', directory, {
      answers: {
        ...answers,
        devDeps: ['jest'],
      },
    })
    expect(JSON.parse(readFile('package.json'))).toMatchSnapshot()
  })
})
