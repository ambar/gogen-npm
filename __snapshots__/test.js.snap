// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template creates with defaults 1`] = `
Array [
  ".gitignore",
  "LICENSE",
  "README.md",
  "lib/index.js",
  "package.json",
]
`;

exports[`template creates with defaults 2`] = `
Object {
  "description": "superb",
  "files": Array [
    "lib",
  ],
  "license": "MIT",
  "name": "dist",
  "scripts": Object {},
  "version": "0.0.0",
}
`;

exports[`template creates with jest 1`] = `
Object {
  "description": "superb",
  "files": Array [
    "lib",
  ],
  "license": "MIT",
  "name": "dist",
  "scripts": Object {
    "prepare": "npm test",
    "test": "jest",
    "test:coverage": "jest --coverage",
    "test:watch": "jest --watch --notify",
  },
  "version": "0.0.0",
}
`;

exports[`template creates with recommended & jest 1`] = `
Object {
  "description": "superb",
  "files": Array [
    "lib",
  ],
  "license": "MIT",
  "name": "dist",
  "scripts": Object {
    "lint": "recommended",
    "lint:fix": "recommended --fix",
    "prepare": "npm run lint && npm test",
    "test": "jest",
    "test:coverage": "jest --coverage",
    "test:watch": "jest --watch --notify",
  },
  "version": "0.0.0",
}
`;

exports[`template creates with recommended 1`] = `
Object {
  "description": "superb",
  "files": Array [
    "lib",
  ],
  "license": "MIT",
  "name": "dist",
  "scripts": Object {
    "lint": "recommended",
    "lint:fix": "recommended --fix",
    "prepare": "npm run lint",
  },
  "version": "0.0.0",
}
`;
